CONTRIBUTING.md

```diff
@@ -45,18 +45,11 @@ during their normal contribution hours.
 
 ## Development
 
-To quickly get up and running, you can use the `scripts/eachdist.py` tool that
```

Comment:
Please document how to achieve this with tox (e.g. I'd like to be able to go-to-definition across projects within VS code/Pylance)

Reply:
Sorry, @Oberon00 I don't understand the following:

> (e.g. I'd like to be able to go-to-definition across projects within VS code/Pylance)

Can you please give me more context? :v:

Reply:
If I just clone the repository, create an empty virtualenv and open the folder in VS Code, it will not be able to resolve an import of, e.g., `Span` from opentelemetry-api while I'm editing a file in `opentelemetry-sdk`.

If I do `pip install -e ./opentelemetry-api`, it will be able to resolve the import.
So one way to do this without eachdist.py is of course to `pip install -e` everything, or everything I'm interested in, manually.

But it would probably possible to replace this with tox somehow. E.g. I saw there is a recent feature with a promising name https://tox.readthedocs.io/en/latest/example/devenv.html.

But what we would need is a tox environment that does a `pip install -e` for each distribution in the project (I think that's where I got eachdist.py's name from originally).

Reply:
We could maybe have a tox ennvironment called `(pyxx-)devenv` that just calls `eachdist.py develop {posargs}` behind the scenes for now. This could be transparently replaced with a solution that does not use the script (I have stated my personal opinion on that removal in #1462)

Reply:
Every time you run `tox`, virtual environments are created in the `.tox` folder and you can activate them, for example, for `tox -e lint`: `source .tox/lint/bin/activate`.

Reply:
I get that, but I was asking was whether there is

>  a tox environment that does a pip install -e for each distribution in the project

Looking at the toxfile now, the `lint` environment seems to do this. Could we mention that here?

Like

> To quickly get up and running, you can run `tox -e lint` and then activate the resulting virtual environment with `. .tox/lint/bin/activate`. This has all distributions in the repository installed in editable mode, as well as the black autoformatter and the pylint linter.

Reply:
Of course, the "quickly" part is hapered a bit by the fact that this environment will run the very slow `pylint`, so maybe a dedicated environment would still be better.

Reply:
Ok, I understand now what you mean. I think that is outside of the scope of this PR, adding that feature to `tox.ini` would be better tracked in a separate issue.

Reply:
Make sense, but maybe suggesting `tox -e lint` would still be better than nothing.

Reply:
It looks like with that devenv feature @Oberon00 shared, you could just update the instructions to say `tox --devenv venv -e lint` and that prevents the commands from running:

> [The --devenv option skips the commands= section of that configured test environment and always sets usedevelop=true for the environment that is created.](https://tox.readthedocs.io/en/latest/example/devenv.html#:~:text=7-,The%20--devenv,If,-you)